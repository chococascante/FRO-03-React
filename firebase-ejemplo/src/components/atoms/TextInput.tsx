import React from "react";

export interface TextInputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
  type: "text" | "password" | "email" | "number";
  placeholder?: string;
  helperText?: string;
}

export const TextInput: React.FC<TextInputProps> = ({
  label,
  type,
  value,
  placeholder,
  helperText,
  ...props
}) => {
  return (
    <label className="flex flex-col mb-6 text-base">
      <span>{label}</span>
      <input placeholder={placeholder} type={type} value={value} {...props} />
      {helperText && <p data-testid="helper-text">{helperText}</p>}
    </label>
  );
};
