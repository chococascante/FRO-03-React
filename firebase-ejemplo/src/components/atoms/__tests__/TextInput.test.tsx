import { render, fireEvent } from "@testing-library/react";
import { TextInput } from "../TextInput";

describe("(Atom) TextInput", () => {
  it.each([undefined, "Hola"])(
    "renders without errors with helperText %s",
    (helperText: string | undefined) => {
      const wrapper = render(
        <TextInput label="" type="text" helperText={helperText} />
      );

      helperText && expect(wrapper.getByText(helperText)).toBeInTheDocument();
    }
  );

  it("deberia cambiar el texto del input", () => {
    const onChange = jest.fn();
    const wrapper = render(
      <TextInput
        label=""
        type="text"
        placeholder="Ingrese su nombre"
        onChange={onChange}
      />
    );
    const input = wrapper.getByPlaceholderText("Ingrese su nombre");
    fireEvent.change(input, { target: { value: "Juan" } });

    expect(wrapper.getByDisplayValue("Juan")).toBeInTheDocument();
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it("deberÃ­a ser igual al snapshot", () => {
    const { container } = render(
      <TextInput label="" type="text" placeholder="Ingrese su nombre" />
    );

    expect(container).toMatchSnapshot();
  });
});
