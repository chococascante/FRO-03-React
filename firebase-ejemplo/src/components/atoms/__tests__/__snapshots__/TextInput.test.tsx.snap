// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Atom) TextInput deber√≠a ser igual al snapshot 1`] = `
<div>
  <label
    class="flex flex-col mb-6 text-base"
  >
    <span />
    <input
      placeholder="Ingrese su nombre"
      type="text"
      value=""
    />
  </label>
</div>
`;
